lib/SampleService/core/validator/builtin.py

```diff
@@ -321,3 +323,67 @@ def _is_num(name, val):
     if val is not None and type(val) != float and type(val) != int:
         raise ValueError(f'Value for {name} parameter is not a number')
     return val
+
+def ontology_has_ancestor(d: Dict[str, Any]) -> Callable[[str, Dict[str, PrimitiveType]], Optional[str]]:
+    '''
+    Build a validation callable that checks that value has valid ontology ancestor term provided
+
+    The 'ontology' parameter is required and must be a string. It is the ontology name.
+
+    The 'ancestor_term' parameter is required and must be a string. It is the ancestor name.
+
+    The 'kbase_endpoint' parameter is required and must be a string. It is kbase endpoint url for getting OntologyAPI service.
+
+    :param d: the configuration map for the callable.
+    :returns: a callable that validates metadata maps.
+    '''
+    _check_unknown_keys(d, {'ontology', 'ancestor_term', 'kbase_endpoint'})
+
+    ontology = d.get('ontology')
+    if not ontology:
+        raise ValueError('ontology is a required paramter')
+    if type(ontology) != str:
+        raise ValueError('ontology must be a string')
+
+    ancestor_term = d.get('ancestor_term')
+    if not ancestor_term:
+        raise ValueError('ancestor_term is a required paramter')
+    if type(ancestor_term) != str:
+        raise ValueError('ancestor_term must be a string')
+
+    endpoint=d.get('kbase_endpoint')
+    if not endpoint:
+        raise ValueError('kbase_endpoint is a required paramter')
+    if type(endpoint) != str:
+        raise ValueError('kbase_endpoint must be a string')
+
+    srv_wiz_url=endpoint.strip('/') + '/service_wizard'
+    oac=None
+    try:
+        oac=OntologyAPI(srv_wiz_url)
+        if oac.status().get('state') != 'OK':
+            raise ValueError('the status of ontology api service is not ok')
+    except:
+        raise ValueError('failed connect to ontology api through kbase_endpoint')
```

Comment:
change to 
```
    except Exception as e:
        raise ValueError('failed connect to ontology api through kbase_endpoint') from e
```
Losing the stack trace and error message could make debugging start up issues really hard